central/imageintegration/service/service_impl.go

```diff
@@ -172,13 +179,27 @@ func (s *serviceImpl) DeleteImageIntegration(ctx context.Context, request *v1.Re
 	if request.GetId() == "" {
 		return nil, errors.Wrap(errox.InvalidArgs, "image integration id must be provided")
 	}
+
+	// Pull the existing integration to determine if should broadcast the delete to sensors.
+	// Do this to avoid sending blind delete messages to potentially many clusters unnecessarily.
+	var doBroadcast bool
+	var name string
+	if ii, exists, err := s.datastore.GetImageIntegration(ctx, request.GetId()); err == nil && exists {
```

Comment:
There's no need for intermediate vars and instead just move this whole check down

Reply:
I believe there is a need, the next line removes the entry from the datastore, if this check is moved down it will never be able to pull the existing value.

Reply:
should we return `err` when it's `!= nil`?

Reply:
I think you can just do the `imageintegration.ValidForSync(ii)` check inside `broadcastDelete` like you do for `broadcastUpdate`, which means you'll pass the whole `ii` and won't need these two vars

Reply:
> should we return err when it's != nil?

I didn't want to change the semantics of the delete. For example, if a delete for a non-existent integration didn't generate an error before, I didn't want to generate an error now.  But could if you feel strongly about it.

> I think you can just do the imageintegration.ValidForSync(ii) check inside broadcastDelete like you do for broadcastUpdate, which means you'll pass the whole ii and won't need these two vars

I like it! Will make that change.